inspectit-oce-documentation/src/docs/asciidoc/_includes/configuration/configuration-basics.adoc

```diff
@@ -0,0 +1,69 @@
+=== Configuration Basics
+
+inspectIT OCE tries to implement the zero-configuration approach but lets you externalize the configuration and influence every bit of its configuration if this is required.
+Internally it uses the Spring-based `PropertySource` order to allow overriding of configuration values.
+Configuration properties are considered in inspectIT OCE in the following order of priority:
+
+. <<Java System properties,Java System properties>>
+. <<OS Environment Variables,OS environment variables>>
+. Supported <<Externalized Configuration Methods,externalized configuration methods>>
+.. <<File-based Configuration,File-based configuration>>
+. <<All Configuration Options,inspectIT OCE defaults>>
+
+The overview of all available inspectIT OCE configuration properties is given in the <<All Configuration Options>> section.
+
+==== Java System Properties
+
+You can pass any configuration property as the Java System property to the Java command that you use to start your Java application.
+For example, to override the service name used to identify your application reporting the performance data, you can change the `inspectit.service-name` property:
+
+[subs=attributes+]
+```bash
+$ java -Dinspectit.service-name=my-custom-service -javaagent:inspectit-oce-agent-{inspectit-oce-version}.jar -jar my-java-program.jar
+```
+
+==== OS Environment Variables
+
+Similar to the Java System properties, inspectIT OCE will also consider all the available operating system environment variables.
+Due to the relaxed bindings, you can use upper case format, which is recommended when using system environment variables.
+
+[subs=attributes+]
+```bash
+$ INSPECTIT_SERVICE_NAME=my-custom-service java -Dinspectit.service-name= -javaagent:inspectit-oce-agent-{inspectit-oce-version}.jar -jar my-java-program.jar
+```
+
+==== Relaxed Bindings
+
+Note that due to the Spring-powered configuration approach, the inspectIT OCE agent uses Spring support for relaxed bindings.
+This means that a property can be specified in different formats depending on the property source.
+As suggested by Spring, the allowed formats are:
+
+[cols="2,3",options="header"]
+.Relaxed binding options for properties
+|===
+|Property |Note
+|```inspectit.service-name```
+|Kebab case, which is recommended for use in .properties and .yml files.
```

Comment:
I would not each case name in its case, e.g. kebab-case, camelCase, snake_case UPPER_CASE